File: kotlin/src/solutions/_003.java
package solutions;

import math.PrimeUtils;

public class _003 {

    public static long largestPrimeFactor() {
        long check = 600851475143L;
        long largestFactor = 1;

        for(long l = 6; l < Math.sqrt(check); l++) {
            if(PrimeUtils.isPrime(l) && check%l == 0) {
                largestFactor = l;
            }
        }
        return largestFactor;
    }
}

File: c/src/003.c
#include <stdio.h>
#include <math.h>

int main(void) {
    long long x = 600851475143;
    long long f = 0;

    for (f = 2; f < sqrt(x); f++) {
        while (0 == x%f && x != f) {
           x /= f;
        }
    }

    printf("%d\n", x);

    return 0;
}

